client/src/modules/Schedule/components/TableView/TableView.tsx

```diff
@@ -122,39 +135,99 @@ export interface TableViewProps {
+    }
+  }, [statusFilter]);
 
-  const filteredData = data
-    .filter(item => (hasStatusFilter(statusFilter, item.status) ? item : null))
-    .filter(event => (tagFilter?.length > 0 ? tagFilter.includes(event.tag) : event));
+  const filteredData = useMemo(() => {
+    return data
+      .filter(item => (hasStatusFilter(statusFilter, item.status) ? item : null))
+      .filter(
+        item =>
+          (combinedFilter?.types?.length ? combinedFilter.types.includes(item.tag) : true) &&
+          (combinedFilter?.statuses?.length ? combinedFilter.statuses.includes(item.status) : true),
+      );
+  }, [combinedFilter, data, statusFilter]);
 
   const filteredColumns = useMemo(
     () =>
       getColumns({
         tagColors: settings.tagColors,
         timezone: settings.timezone,
-        tagFilter,
+        combinedFilter,
         filteredInfo,
+        currentTabKey: statusFilter,
       }).filter(column => {
         const key = (column.key as ColumnKey) ?? ColumnKey.Name;
         return CONFIGURABLE_COLUMNS.includes(key) ? !settings.columnsHidden.includes(key) : true;
       }),
-    [settings.columnsHidden, settings.timezone, settings.tagColors, statusFilter, tagFilter],
+    [settings.columnsHidden, settings.timezone, settings.tagColors, statusFilter, combinedFilter],
   );
   const columns = filteredColumns as ColumnsType<CourseScheduleItemDto>;
 
   const handleTagClose = (removedTag: string) => {
-    setTagFilter(tagFilter.filter(t => t !== removedTag));
+    const [removedTagName, ...removedTagValueParts] = removedTag.split(':');
+    const removedTagValue = removedTagValueParts.join(':');
+
+    const filterTags = combinedFilter.filterTags?.filter(filter => filter !== removedTag);
+
+    switch (removedTagName) {
+      case ColumnName.Type:
+        setCombinedFilter({
+          ...combinedFilter,
+          types: combinedFilter.types.filter(tag => tag !== REVERSE_TAG_NAME_MAP[removedTagValue.trim()]),
+          filterTags,
+        });
+        break;
+      case ColumnName.Status:
+        setCombinedFilter({
+          ...combinedFilter,
+          statuses: combinedFilter.statuses.filter(tag => tag !== removedTagValue.trim().toLowerCase()),
+          filterTags,
+        });
+        break;
+      default:
+        message.error('An error occurred. Please try again later.');
+        break;
+    }
   };
 
   const handleClearAllButtonClick = () => {
-    setTagFilter([]);
+    setCombinedFilter({ types: [], statuses: [], filterTags: [] });
+  };
+
+  const handleTableChange = (_: any, filters: Record<ColumnKey, FilterValue | string[] | null>) => {
+    const combinedFilter: CombinedFilter = {
+      types: filters.type?.map(tag => tag.toString()) ?? [],
+      statuses: filters.status?.map(status => status.toString()) ?? [],
+    };
+
+    combinedFilter.filterTags = [
+      ...combinedFilter.types.map(tag => `${ColumnName.Type}: ${TAG_NAME_MAP[tag as CourseScheduleItemDto['tag']]}`),
```

Comment:
no need to use label as value and later split by ':' and using reverse map.
For example, consider using the following structure instead of just string:

```typescript
type FilterTag = {
  label: string
  value: string 
  tagType: 'type' | 'status'
}
```

Reply:
@apalchys could you please check if I understand and implement your suggestion correctly?
I was thinking about making `combinedFilter` as an array of `FilterTag` items, but that way I believe we are forced to filter that array all the time to separate items by type

Reply:
it's better imno. thanks.